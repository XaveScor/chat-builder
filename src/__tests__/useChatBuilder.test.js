import {createPage, questionPhrase, simplePhrase, input, runConforms, createProps, Stop, useChatBuilder} from '..'
import * as React from 'react';
import * as renderer from 'react-test-renderer';

it('init props', done => {
    const value = {a: 'incorrect props sharing'}
    const shared = createProps(value)
    const page = createPage({
        props: shared,
    })

    page.use((_, props) => {
        return {
            steps: [
                {
                    ...simplePhrase,
                    input,
                    question: props.a
                }
            ],
            nextPage: Stop,
        }
    })

    const Component = () => {
        const Chat = useChatBuilder(page)

        return <Chat a='correct props sharing' />
    }

    const tree = renderer
        .create(<Component />)


    setTimeout(() => {
        const res = tree.toJSON()
        expect(res).toMatchSnapshot()
        done()
    }, 1000)
})