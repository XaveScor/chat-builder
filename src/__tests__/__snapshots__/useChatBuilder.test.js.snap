// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init props 1`] = `
<p
  className="robot"
>
  correct props sharing
</p>
`;
